cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -89,6 +96,22 @@ func SetDefaults_MasterConfiguration(obj *MasterConfiguration) {
 	if obj.Etcd.DataDir == "" {
 		obj.Etcd.DataDir = DefaultEtcdDataDir
 	}
+
+	if obj.Etcd.SelfHosted.ClusterSize == 0 {
+		obj.Etcd.SelfHosted.ClusterSize = DefaultEtcdClusterSize
+	}
+
+	if obj.Etcd.SelfHosted.EtcdVersion == "" {
+		obj.Etcd.SelfHosted.EtcdVersion = "v" + constants.DefaultEtcdVersion
```

Comment:
sanitize to always include the `v`; even if you give the value yourself?

Reply:
Actually I just noticed that `3.1.8` format seems to be preferred, but `v` will also work. I'm gonna remove prepending `v` and let the user decide. I think validation of input is better handled by the operator TBH.